var/spack/repos/builtin/packages/geant4/package.py

```diff
@@ -172,6 +204,9 @@ def cmake_args(self):
                 '-DQT_QMAKE_EXECUTABLE=%s' %
                 spec['qt'].prefix.bin.qmake)
 
+        if '+vtk' in spec:
+            options.append('-DGEANT4_USE_VTK=ON')
```

Comment:
Does define from variant not work when the variant is optional? I would use that shortcut and wrap in a 'if spec.version >= 11'.

Reply:
`define_from_variant` works fine here indeed. I was using local context to guide style and originally I had included a specification of the prefix for vtk but that turned out not to be necessary. I'll simplify this as:
```suggestion
        options.append(self.define_from_variant('GEANT4_USE_VTK=ON','vtk'))
```